### PR TITLE
rejigger how initContainers are mutated to not forget VolumeMounts

### PR DESCRIPTION
**Pull request by byxorna**
# What and why?

A small refactor to catch a bug managing initContainers - we were not injecting VolumeMounts into existing InitContainers in the AdmissionControlRequest. I have added tests to validate this behavior now, and did some slight changes to make sure we are injecting env vars+volumeMounts into initContainers.

@komapa @alex-laties @defect 

# Testing Steps

Run unit tests

- [x] Added unit tests for this feature (`make test`)

# Reviewers

Required reviewers: @byxorna
Request reviews from other people you want to review this PR in the "Reviewers" section on the right.

> :warning: this PR must have at least 2 thumbs from the [MAINTAINERS.md](/MAINTAINERS.md) of the project before merging!

### CODE REVIEW

**Comment by komapa** [comment]
```
--- FAIL: TestGoodConfigs (0.00s)
366    config_test.go:227: expected 0 Envs loaded from test/fixtures/sidecars/maxmind.yaml but got 1
```

**Comment by byxorna** [comment]
@komapa @alex-laties another round, if you please? Travis is a bit flaky :)

File: pkg/server/webhook.go [hunk 1]
```diff
@@ -231,34 +231,39 @@ func addContainers(target, added []corev1.Container, basePath string) (patch []p
...
+			// if any of the existing volumes have the same name as test.Name, skip
+			// injecting it
+			if v.Name == add.Name {
+				return true
+			}
+		}
+		return false
+	}
 	for _, add := range added {
-		value = add
```
Comment (byxorna) [inline pkg/server/webhook.go]: This was causing issues with duplicated volumes

File: pkg/server/webhook.go [hunk 2]
```diff
@@ -348,21 +353,25 @@ func setServiceAccount(initContainers []corev1.Container, containers []corev1.Co
...
+			}
+		}
+		return false
+	}
 	mutatedContainers := []corev1.Container{}
 	for _, c := range containers {
 		for _, newEnv := range envs {
 			// check each container for each env var by name.
 			// if the container has a matching name, dont override!
-			skip := false
```
Comment (byxorna) [inline pkg/server/webhook.go]: this logic was confusing and the control flow also was suspect, so i moved it into a helper func

**Pull request merged**